src/core/guards/__tests__/rate-limit.guard.spec.ts

```
@@ -0,0 +1,336 @@
+    it('should reject when rate limit is exceeded', async () => {
+      process.env.RATE_LIMIT_AUTH_STRICT_MAX = '1';
+      process.env.RATE_LIMIT_AUTH_WINDOW_MS = '60000';
+
+      const guardWithLowLimit = new AuthRateLimitGuard();
+
+      const result1 = await guardWithLowLimit.canActivate(mockContext);
+      expect(result1).toBe(true);
+
+      expect(process.env.RATE_LIMIT_AUTH_STRICT_MAX).toBe('1');
+
+      delete process.env.RATE_LIMIT_AUTH_STRICT_MAX;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)